### ISSUE TITLE
Batch Opening and Saving Throttle

### ISSUE THREAD

**Issue by coryasilva** [KamasamaK/vscode-cflint#1]
When using the "Save All" command or find and replacing across many files (which in turn opens every file) this extension attempts to lint.  While, I like the linting onOpen and onSave, it would be nice if it could throttle itself and check if the file is still open before running.

Consider this desired effect:
- Open 50 files
  - lint a max of config.throttleLimit (default: 5)
- Save 50 files
  - lint a max of config.throttleLimit (default: 5)

I did this with a 200+ file batch the other day, it was painful.  lol

**Comment by KamasamaK** [comment]
@coryasilva I have attempted a solution for this. Please verify.

**Comment by coryasilva** [comment]
Okay I have repeated the process with 36 files.  
- Find and Replace in directory `t=0s`
- 36 files opened (with replaced text) `t~2s`
- 36 files saved (File > Save All) `t~4s`
- 36 files closed (Command + W hit manually 36 times) `t~30s`
- Last linter process ran `t~300s`

While it the maxSimultaneousLints seemed to be in effect, a lot of lint processes ran pinning the CPU for about 5 minutes after all the files were closed.  That said my computer was responsive though!

![image](https://user-images.githubusercontent.com/2672972/31092059-08cb6ca4-a762-11e7-8f80-f00ee55bb9de.png)
